x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherPluginTests.java

```diff
@@ -97,4 +103,32 @@ public void testThreadPoolSize() {
                 .build();
         assertThat(Watcher.getWatcherThreadPoolSize(noDataNodeSettings), is(1));
     }
+
+    public void testReload() {
+        Settings settings = Settings.builder()
+            .put("xpack.watcher.enabled", true)
+            .put("path.home", createTempDir())
+            .build();
+        NotificationService mockService = mock(NotificationService.class);
+        Watcher watcher = new Watcher(settings);
+        watcher.reloadableServices.clear();
+        watcher.reloadableServices.add(mockService);
+
+        verify(mockService, times(0)).reload(settings);
+        watcher.reload(settings);
+        verify(mockService, times(1)).reload(settings);
+    }
+
+    public void testReloadDisabled() {
+        Settings settings = Settings.builder()
+            .put("xpack.watcher.enabled", false)
+            .put("path.home", createTempDir())
+            .build();
+        NotificationService mockService = mock(NotificationService.class);
+        Watcher watcher = new Watcher(settings);
+
+        verify(mockService, times(0)).reload(settings);
+        watcher.reload(settings);
+        verify(mockService, times(0)).reload(settings);
```

Comment:
there is a `verifyNoInteractions()` method, that makes this test a bit easier to read